system/jlib/jtime.hpp

```diff
@@ -83,7 +83,7 @@ class jlib_decl CDateTime
     void setFromFILETIME(__int64 fileTime);
     void setTimeStamp(timestamp_type ts);
 
-    void setString(char const * str, char const * * end = NULL, bool local = false); // Sets to date and time given as yyyy-mm-ddThh:mm:ss[.nnnnnnnnn]
+    void setString(char const * str, char const * * end = NULL, bool local = false); // Sets to date and time given as yyyy-mm-dd[T ]hh:mm:ss[.nnnnnnnnn]
```

Comment:
Not convinced by this change.
CDateTime::getString always returns a formated string with a 'T' separator.

This is being changed, because TimeStarted in dfu workunits is being recorded with a space instead of a 'T' (so not currently a CDateTime formatted string).

Why is the service setting TimeStarted with a non CDateTime format at the moment, i.e. in this code in (DeepAssign):
```
        CDateTime startAt;
...
        startAt.getDateString(tmpstr);
        tmpstr.append(" ");
        startAt.getTimeString(tmpstr);
        dest.setTimeStarted(tmpstr.str());
```
it could be:
```
        CDateTime startAt;
...
        startAt.getString(tmpstr);
        dest.setTimeStarted(tmpstr.str());
```

It's also inconsistent with the other place that sets TimeStarted in (onGetDFUWorkunits), which does build a regular CDateTime string format : 
```
                        CDateTime startAt, stopAt;
...
                        StringBuffer s;
                        resultWU->setTimeStarted(startAt.getString(s));
```

I think both should call setTimeStarted with a regular CDateTime format (i.e. via getString).

Reply:
I don't know why onGetDFUWorkunits doesn't build a regular CDateTime string format.   I was thinking that changing the format could break something.  I'm not sure if anything is relying on the format being without the 'T' separator.

Do you think we should be, update the onGetDFUWorkunit to return the CDateTime format?

Reply:
By the way,  I don't actually see any problems with supporting both formats.